drivers/i2c/i2c_emul.c

```diff
@@ -74,8 +74,8 @@ static int i2c_emul_get_config(const struct device *dev, uint32_t *dev_config)
 	return 0;
 }
 
-static int i2c_emul_transfer(const struct device *dev, struct i2c_msg *msgs,
-			     uint8_t num_msgs, uint16_t addr)
+static int i2c_emul_transfer(const struct device *dev, struct i2c_msg *msgs, uint8_t num_msgs,
```

Comment:
Why are you changing this perfectly formatted piece of code here (and on other places)? These changes are not necessary at all.

Reply:
[Zephyr uses clang-format](https://github.com/zephyrproject-rtos/zephyr/blob/main/.clang-format) to properly format code. My understanding is the project is working on making this more consistent and enforced on future changes.

Making code consistently formatted is necessary to improve readability and future diffs easier to parse.

So this change is certainly necessary. Are you suggesting it belongs in its own PR?

Reply:
No. From [Contribution Guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style), "The [clang-format tool](https://clang.llvm.org/docs/ClangFormat.html) can be helpful to quickly reformat large amounts of new source code". Is is not new code., most of the code is perfectly formatted. It conform to our coding style, which ->` In general, follow the [Linux kernel coding style](https://kernel.org/doc/html/latest/process/coding-style.html),` -> "The preferred limit on the length of a single line is 80 columns."

Reply:
Then it looks like the Code Style section of the contribution guidelines doc is out-of-date. The [clang-format configuration change increasing the line length to 100](https://github.com/zephyrproject-rtos/zephyr/pull/46952) strongly suggests this is the case since there's no sensible reason to have two different styles within the same project unless the project is slowly migrating to from one to the other; that is my understanding here.

[This was the last change to the contribution guidelines](https://github.com/zephyrproject-rtos/zephyr/commit/c7e4388b1dc5a42f171b87af70d2c033b4d9b917) which happened before the `clang-format` config change, which probably should've happened in the [clang-format config change PR](https://github.com/zephyrproject-rtos/zephyr/pull/46952).

Between the contribution guidelines or the clang-format configuration, I'd absolutely consider the clang-format configuration as the source of truth since docs grow stale.

As for applying `clang-format` to "new source code": here `clang-format` was only applied to files changed within this PR. It'd be quite tedious to only apply `clang-format` specifically to new code regions and this would make the code within a source file inconsistently formatted.

Requesting a review from @nashif to comment on this here.

Reply:
None of this justifies the change to the perfectly formatted code. I do not see different styles, I see imperfection of clang-format. Also, our coding styles are not so complicated that an average contributor could not follow them.

Reply:
Marking this as resolved, @jfischer-no I understand your concern, but the issue here is that if files are formatted differently, then when the next developer is going in and making future changes they will first have to learn the style of the current file in order to match it. The alternative is to have sections of code with different formats (which I think all of us will agree is not acceptable). @aaronemassey 's approach here is reasonable, running clang-format in a separate commit first, then applying changes to make review easier.

Reply:
No I strongly disagree. We do not refactor the code in the project that complies with coding style. clang-format reformatted code does not match our established style. And here currently you are enforcing different coding style.

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/30426

Reply:
> #30426

Well maximum length in checkpatch does not apply here (and coding-style is already quoted above). The issue is also how clang-format formats the parameters and arguments of a function. There will always be unnecessary changes, and so clang-format is not suited to do formatting without manual post-processing.